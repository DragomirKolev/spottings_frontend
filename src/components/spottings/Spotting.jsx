import React, { useState } from 'react';
import { createUseStyles } from 'react-jss';
import arrowUpSrc from '../../img/arrow_up.svg';
import arrowDownSrc from '../../img/arrow_down.svg';

import { GLOBAL_SPACING, styles } from '../../styles/common';

const useStyles = createUseStyles({
  row: {
    paddingBottom: GLOBAL_SPACING,
    borderBottom: styles.border('green'),
    cursor: 'pointer',
    display: 'flex',
  },
  rowContent: {
    marginLeft: GLOBAL_SPACING * 2,
  },
  rowItem: {
    display: 'flex',
    flexDirection: 'row'
  }
});

const getUserFacingTimestamp = (date) => {
  return `${date.getDate()}/${1 + date.getMonth()}/${date.getFullYear()} 
  ${date.getHours()}:${date.getMinutes()}`
}

const Spotting = ({ spotting }) => {
  const classes = useStyles();

  const [open, setOpen] = useState(false);
  const src = open ? arrowUpSrc : arrowDownSrc;
  const userFacingDate = getUserFacingTimestamp(new Date(spotting.date_time));

  return (
    <div onClick={() => setOpen(!open)} className={classes.row}>
      <img className={classes.arrow} src={src} alt="logo" width='24px' height='24px' />
      <div className={classes.rowContent}>
        <div className={classes.rowItem}> Date and Time : {userFacingDate}</div> 
        <div className={classes.rowItem}> Bird Type : {spotting.bird_type}</div> 
        <div className={classes.rowItem}> Confident : {spotting.confident ? 'Yes' : 'No'}</div>
          {open && <>
              <div className={classes.rowItem}> Name : {spotting.name}</div>
              <div className={classes.rowItem}> Place : {spotting.location} </div>
          </>}
      </div>
    </div>
  );
};

export default Spotting;