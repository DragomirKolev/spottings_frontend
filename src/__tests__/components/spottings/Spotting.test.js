import React from 'react';
import renderer from 'react-test-renderer';

import Spotting from '../../../components/spottings/Spotting';

it('renders Spotting', () => {
  const spotting = {
    'date_time': '2020-01-19T14:17:51.495Z',
	  'location': 'in the middle of nowhere',
	  'bird_type': 'pidgeon',
	  'name': 'George Paul',
	  'confident': true
  }
  const tree = renderer
    .create(<Spotting spotting={spotting} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});