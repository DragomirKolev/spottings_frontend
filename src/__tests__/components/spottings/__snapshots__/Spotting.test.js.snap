// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Spotting 1`] = `
<div
  className="row-0-2-1"
  onClick={[Function]}
>
  <img
    alt="logo"
    height="24px"
    src="arrow_down.svg"
    width="24px"
  />
  <div
    className="rowContent-0-2-2"
  >
    <div
      className="rowItem-0-2-3"
    >
       Date and Time : 
      19/1/2020 
  15:17
    </div>
    <div
      className="rowItem-0-2-3"
    >
       Bird Type : 
      pidgeon
    </div>
    <div
      className="rowItem-0-2-3"
    >
       Confident : 
      Yes
    </div>
  </div>
</div>
`;
